mempool/src/shared_mempool.rs

```diff
@@ -479,35 +471,43 @@ async fn process_consensus_request<V>(
 
             let transactions = txns.drain(..).map(SignedTransaction::into).collect();
 
-            MempoolResponse::GetBlockResponse { transactions }
+            (MempoolResponse::GetBlockResponse { transactions }, callback)
         }
-        MempoolRequest::CommitTransactionsRequest {
-            transactions,
-            block_timestamp_usecs,
+        MempoolRequest::CommitNotification {
+            old_version,
+            new_version,
+            callback,
         } => {
-            let mut pool = smp.mempool.lock().unwrap();
-            for transaction in transactions.iter() {
-                pool.remove_transaction(
-                    &transaction.sender,
-                    transaction.sequence_number,
-                    transaction.is_rejected,
-                );
-            }
-
-            if block_timestamp_usecs > 0 {
-                pool.gc_by_expiration_time(Duration::from_micros(block_timestamp_usecs));
+            let committed_txns = smp
+                .storage_read_client
+                .get_transactions(
+                    old_version,
+                    new_version - old_version + 1,
+                    new_version,
+                    false,
+                )
+                .await
+                .expect("[shared mempool] failed to read from storage");
+            let txns = committed_txns.transactions;
+            let mut mempool = smp
+                .mempool
+                .lock()
+                .expect("[shared mempool] failed to get mempool lock");
+            for txn in txns {
+                if let Ok(txn) = txn.as_signed_user_txn() {
+                    mempool.remove_transaction(&txn.sender(), txn.sequence_number(), false);
+                }
             }
-
-            MempoolResponse::CommitTransactionsResponse {}
+            (MempoolResponse::CommitResponse {}, callback)
```

Comment:
maybe I'm just missing that part, but where's handling new on-chain timestamp
we need to do GC in Mempool based on that value